server/data/prisonerClient.ts

```diff
@@ -11,9 +12,18 @@ export default class PrisonerClient {
     this.restClient = new RestClient('prisonerClient', config.apis.prisonerOffenderSearch as ApiConfig, token)
   }
 
-  async find(prisonNumber: string): Promise<Prisoner> {
-    return (await this.restClient.get({
-      path: prisonerOffenderSearchPaths.prisoner.show({ prisonNumber }),
-    })) as Prisoner
+  async find(prisonNumber: string, caseloadIds: Array<Caseload['caseLoadId']>): Promise<Prisoner | null> {
+    const prisoners: Array<Prisoner> = (await this.restClient.post({
+      data: {
+        prisonIds: caseloadIds,
+        prisonerIdentifier: prisonNumber,
+      },
+      path: prisonerOffenderSearchPaths.prisoner.search({}),
+    })) as Array<Prisoner>
+
+    if (!prisoners.length) {
+      return null
+    }
+    return prisoners[0]
```

Comment:
Can probably be `return prisoners[0] || null` but I'm not sure if that's as readable